### PR TITLE
Add new download file API in the doc as well restriction based on number of upload step a pipeline can have

### PR DESCRIPTION
**Pull request by krishi0408**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

Screenshot 2024-12-10 at 11 42 43 AM

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/platform/variables-and-expressions/runtime-input-usage.md [hunk 1]
```diff
@@ -471,8 +471,19 @@ To get the exact uploaded file name, navigate to the `Output` tab of the upload
 To download the uploaded file, use the following cURL command:
 
 ```
-curl --location --request GET 'https://app.harness.io/gateway/pipeline/api/input-file/file/PLAN_EXECUTION_ID?accountIdentifier=ACCOUNT_IDENTIFIER&nodeExecutionId=NODE_EXECUTION_ID&fileName=FILE_NAME' \
+curl --location 'https://app.harness.io/gateway/pipeline/api/input-file/download-file?accountIdentifier=ACCOUNT_IDENTIFIER&filePath=accountId&filePath=pipelineExecutionId/nodeExecutionId/fileName.extension' \
 --header 'x-api-key: PAT_TOKEN'
```
Comment (av3esha) [inline docs/platform/variables-and-expressions/runtime-input-usage.md:475]: lets keep both the end points

Comment (krishi0408) [reply docs/platform/variables-and-expressions/runtime-input-usage.md]: Sure @av3esha , added both in the docs.

File: docs/platform/variables-and-expressions/runtime-input-usage.md [hunk 2]
```diff
@@ -468,20 +468,36 @@ For example, if the original file name is `input.txt`, it will be stored as `inp
...
-To download the uploaded file, use the following cURL command:
+To download the uploaded file, use the following CURL command:
 
 ```
 curl --location --request GET 'https://app.harness.io/gateway/pipeline/api/input-file/file/PLAN_EXECUTION_ID?accountIdentifier=ACCOUNT_IDENTIFIER&nodeExecutionId=NODE_EXECUTION_ID&fileName=FILE_NAME' \
 --header 'x-api-key: PAT_TOKEN'
 ```
 
+```
+curl --location 'https://app.harness.io/gateway/pipeline/api/input-file/download-file?accountIdentifier=ACCOUNT_IDENTIFIER&filePath=accountId&filePath=pipelineExecutionId/nodeExecutionId/fileName.extension' \
```
Comment (av3esha) [inline docs/platform/variables-and-expressions/runtime-input-usage.md:479]: the curl is wrong:

`https://qa.harness.io/gateway/pipeline/api/input-file/download-file?accountIdentifier=px7xd_BFRCi-pfWPYXVjvw&filePath=FILE_PATH`

filePath is repeated in the curl which is added to the doc.
also, just mention the filePath in the url, we can add the format of the file path and where to get it from (i.e. in the step execution logs and in the output variables)

Comment (krishi0408) [reply docs/platform/variables-and-expressions/runtime-input-usage.md]: corrected the curl

File: docs/platform/variables-and-expressions/runtime-input-usage.md [hunk 3]
```diff
@@ -468,20 +468,36 @@ For example, if the original file name is `input.txt`, it will be stored as `inp
 To get the exact uploaded file name, navigate to the `Output` tab of the upload step execution, where the full file name can be found.
 
 #### Retrieve Uploaded File Using cURL
-To download the uploaded file, use the following cURL command:
+To download the uploaded file, use the following CURL command:
 
 ```
-curl --location --request GET 'https://app.harness.io/gateway/pipeline/api/input-file/file/PLAN_EXECUTION_ID?accountIdentifier=ACCOUNT_IDENTIFIER&nodeExecutionId=NODE_EXECUTION_ID&fileName=FILE_NAME' \
+curl --location 'https://app.harness.io/gateway/pipeline/api/input-file/file/PLAN_EXECUTION_ID?accountIdentifier=ACCOUNT_IDENTIFIER&nodeExecutionId=NODE_EXECUTION_ID&fileName=FILE_NAME' \
 --header 'x-api-key: PAT_TOKEN'
```
Comment (av3esha) [inline docs/platform/variables-and-expressions/runtime-input-usage.md:475]: discussed it internally, we should remove this.
Bringing clarity as to which end point to use when , will be difficult.

File: docs/platform/variables-and-expressions/runtime-input-usage.md [hunk 4]
```diff
@@ -468,20 +468,36 @@ For example, if the original file name is `input.txt`, it will be stored as `inp
...
+
+Parameters:
+
+- **accountIdentifier**: Required – Specifies the account ID.
+
+- **filePath**: Required – The file path should follow this format:
+
+```
+accountId/runtimeFileInputData/pipelineExecutionId/nodeExecutionId/fileName.extension
+```
```
Comment (av3esha) [inline docs/platform/variables-and-expressions/runtime-input-usage.md:491]: can you add here that the file path can be fetched form the step execution logs or directly accessible via the output variables.

Comment (krishi0408) [reply docs/platform/variables-and-expressions/runtime-input-usage.md:491]: hey @av3esha added in notes below

File: docs/platform/variables-and-expressions/runtime-input-usage.md (hunk 3 above)
Comment (krishi0408) [reply docs/platform/variables-and-expressions/runtime-input-usage.md:475]: Sure

File: docs/platform/variables-and-expressions/runtime-input-usage.md (hunk 4 above)
Comment (krishi0408) [reply docs/platform/variables-and-expressions/runtime-input-usage.md:486]: Previously added in notes below, moved it in the same section

**Review by SudarshanPurohit — APPROVED** [review]

**Review by av3esha — APPROVED** [review]
lgtm

**Review by adivishy1 — APPROVED** [review]

**Pull request merged**